Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -314,23 +324,46 @@ void OpenALStream::SoundLoop()
     // many silence samples.  These do not need to be timestretched.
     if (rate > 0.10)
     {
-      soundTouch.setTempo(rate);
-      if (rate > 10)
+      if (use_timestretching)
+      {
+        soundTouch.setTempo(rate);
+      }
+      else
+      {
+        alSourcef(ui_source, AL_PITCH, static_cast<ALfloat>(rate));
```

Comment:
Yeah, the terminology is confusing. Pitch-shifting changes the pitch but not the tempo; time-stretching changes the tempo but not the pitch; and resampling/frequency-shifting changes both at the same time.

So the Mixer class does frequency-shifting internally (unless you ask it not to, with that boolean third parameter) and SoundTouch does time-stretching/pitch-shifting (which are the same: if you have a time-stretcher, you can combine that with a resampler to get a pitch-shifter, and vice-versa).